invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/DocumentDetails/components/DocumentAccessRestrictions/DocumentAccessRestrictions.js

```diff
@@ -0,0 +1,183 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Button,
+  Header,
+  Icon,
+  Modal,
+  Grid,
+  Search,
+  Segment,
+  Label,
+} from 'semantic-ui-react';
+import { debounce } from 'lodash';
+import './DocumentAccessRestrictions.scss';
+
+const searchResultsRenderer = ({ title, description, price }) => (
+  <div key="content" className="content">
+    <div className="price">{price}</div>
+    <div className="title">{title}</div>
+    <div className="description">{description}</div>
+  </div>
+);
+
+export default class DocumentAccessRestrictions extends Component {
+  constructor(props) {
+    super(props);
+    this.searchPatrons = props.searchPatrons;
+    this.setRestrictionsOnDocument = props.setRestrictionsOnDocument;
+    this.pid = props.document.document_pid;
+    this.searchPatronsDebounced = debounce(this.searchPatrons, 500);
+  }
+
+  state = { isModalOpen: false, currentAccessList: this.props.accessList };
+
+  toggleModal = () =>
+    this.setState({
+      isModalOpen: !this.state.isModalOpen,
+      currentAccessList: this.props.accessList,
+    });
+
+  toggleButton = () =>
+    this.setState({ isButtonClicked: !this.state.isButtonClicked });
+
+  resetComponent = () => this.setState({ isLoading: false, value: '' });
+
+  handleResultSelect = (e, { result }) => {
+    this.setState(state => {
+      const currentAccessList = new Set([
+        ...Array.from(state.currentAccessList),
+        result.title,
+      ]);
+      return {
+        currentAccessList,
+        value: result.title,
+      };
+    });
+    this.resetComponent();
+  };
+
+  handleRemoveSelection = email => e => {
+    this.setState(state => {
+      const currentAccessList = new Set(
+        Array.from(state.currentAccessList).filter(item => email !== item)
+      );
+      return {
+        currentAccessList,
+      };
+    });
+  };
+
+  renderCurrentSelection = () => {
+    const labels = Array.from(this.state.currentAccessList).map(email => (
+      <div className="current-selection-labels" key={email}>
+        <Label as="a" color="blue">
+          {email}
+          <Icon name="delete" onClick={this.handleRemoveSelection(email)} />
+        </Label>
+      </div>
+    ));
+    return labels ? <div>{labels}</div> : null;
+  };
+
+  handleSearchChange = (e, { value }) => {
+    this.setState({ isLoading: true, value });
+
+    setTimeout(() => {
+      if (this.state.value.length < 1) return this.resetComponent();
+
+      if (this.state.value.length > 1)
```

Comment:
There is a strange behaviour when search for something (e.g patron) and then you delete everything and type `a` you are still getting the patron results as no new search was triggered because if the `if (this.state.value.length < 1)`. In general I would propose to discuss again about of the workflow of this component and its different states because I think it's not consistent :)